### PR TITLE
Bullet Type Visual Indicator

### LINKED ISSUE #1741: Make bullet types easily recognizable while in-air
So you know if someone's shooting at you with a rubber bullet or a real bullet.

### PR DESCRIPTION
**Pull request by MaloTV**
## About the PR
Changes Bullet sprites according to their Types (Normal, Practice, Rubber)
Closes #1741 

## Why / Balance
Added so players know which bullet they are being shot with by your local Greytides and Secoffs

## Media
Practice Bullet
image

https://github.com/space-wizards/space-station-14/assets/70515562/a0a7cb9f-1be2-41ad-8ca1-b24cc5baa9ef

Rubber Bullet
image

https://github.com/space-wizards/space-station-14/assets/70515562/d9eb98ca-2132-49ed-873a-5f43d118989c

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:Malo
- tweak: Bullet types (rubber, practice, normal) can now be visually distinguished from each other!

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 1]
```diff
@@ -95,6 +95,12 @@
   parent: BaseBullet
   noSpawn: true
   components:
+  - type: Sprite
+    noRot: false
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:99]: You don't need to re-specify the same values as parent.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 2]
```diff
@@ -106,6 +112,12 @@
...
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:116]: ditto here for noRot.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 3]
```diff
@@ -95,6 +95,12 @@
...
+    sprite: Objects/Weapons/Guns/Projectiles/projectiles2.rsi
+    layers:
+    - state: practice
+      shader: unshaded
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:103]:
```suggestion
      shader: unshaded
```

Please remove unnecessary trailing whitespace.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 4]
```diff
@@ -106,6 +112,12 @@
   parent: BaseBullet
   noSpawn: true
   components:
+  - type: Sprite
+    noRot: false
+    sprite: Objects/Weapons/Guns/Projectiles/projectiles2.rsi
+    layers:
+    - state: rubber
+      shader: unshaded
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:120]:
```suggestion
      shader: unshaded
```

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json [hunk 5]
```diff
@@ -24,6 +24,24 @@
           0.05
         ]
       ]
+    },
+	{
+      "name": "rubber",
```
Comment (metalgearsloth) [inline Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json:29]: Please don't mix tabs and spaces, if you use VS or Rider it should handle this for you.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Flareguy** [comment]
I'm fine with the rubber bullets, however practice is kind of weird and the orange doesn't feel very fitting. I would just suggest making it like the rubber ones, making it appear mostly like a normal bullet except for the tip (which should probably be red, like the mag.)

**Comment by MaloTV** [comment]
> I'm fine with the rubber bullets, however practice is kind of weird and the orange doesn't feel very fitting. I would just suggest making it like the rubber ones, making it appear mostly like a normal bullet except for the tip (which should probably be red, like the mag.)

practice

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json (hunk 5 above)
Comment (MaloTV) [reply Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json:29]: Please explain for i do not understand what you mean?

Comment (Darkenson) [reply Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json:29]: On line 27, before the "}," there are four spacebar spaces.
On line 28, before the "{" there is one tab button space.
On line 29 you use spacebar spaces again.

Sloth would like it if you consistently use one of them. Visual Studio or Rider do this automatically.

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json [hunk 6]
```diff
@@ -24,6 +24,24 @@
           0.05
         ]
       ]
+    },
+	{
```
Comment (Darkenson) [inline Resources/Textures/Objects/Weapons/Guns/Projectiles/projectiles2.rsi/meta.json:28]: Line 28 still has a tab when everything else has spaces.

**Review by Darkenson — CHANGES_REQUESTED** [review]
Fix line 28 to have spacebar spaces and that should be it.

**Review by deathride58 — APPROVED** [review]

**Pull request merged**